pkg/server/check.go

```diff
@@ -60,13 +60,12 @@ func (s *Server) Check(ctx context.Context, req *openfgav1.CheckRequest) (*openf
 	}
 
 	checkQuery := commands.NewCheckCommand(
-		s.checkDatastore,
+		s.datastore,
 		s.checkResolver,
-		typesys,
 		commands.WithCheckCommandLogger(s.logger),
 		commands.WithCheckCommandMaxConcurrentReads(s.maxConcurrentReadsForCheck),
 		commands.WithCheckCommandResolveNodeLimit(s.resolveNodeLimit),
-		commands.WithCacheController(s.cacheController),
+		commands.WithCheckCommandCache(s.cacheController, shouldCacheIterators(s), s.singleflightGroup, s.checkCache, s.checkIteratorCacheMaxResults, s.checkQueryCacheTTL),
```

Comment:
Why are we introducing this change?

Reply:
So that when a request comes in, I can decide what datastore decorators I need to instantiate; specifically with this line I can decided whether we should call `NewCachedDatastore` or not.